source/javascripts/services/request-service.ts

```diff
@@ -0,0 +1,158 @@
+import { getAppSlug } from "./app-service";
+import StringService from "./string-service";
+import { Logger } from "./logger";
+import { StatusType } from "@datadog/browser-logs";
+
+enum RequestServiceMode {
+	Website = "website",
+	Cli = "cli"
+}
+
+interface RequestServiceConfigureOptions {
+	mode?: string;
+	appSlug?: string;
+	logger?: Logger;
+}
+
+class RequestService {
+	public mode = "";
+	public appSlug = "";
+	private logger: Logger | undefined;
+
+	constructor(logger: Logger) {
+		this.configure({ logger });
+	}
+
+	public configure({ mode, appSlug, logger }: RequestServiceConfigureOptions = {}): void {
+		this.mode = mode || (this.modeFromEnvVars() as string);
+		this.appSlug = appSlug || (getAppSlug() as string);
+		this.logger = logger;
+	}
+
+	public async getAppConfigYML(
+		abortedPromise: Promise<undefined>
+	): Promise<string | Error | { bitrise_yml: string; error_message: Error }> {
+		const controller = new AbortController();
+		if (abortedPromise) {
+			abortedPromise.then(() => {
+				controller.abort();
+			});
+		}
+
+		let requestURL = "";
+
+		switch (this.mode) {
+			case RequestServiceMode.Website:
+				requestURL = StringService.stringReplacedWithParameters(window["routes"].website.yml_get, {
```

Comment:
Not sure if i agree with collecting all endpoint paths in a separate file. I don't think we ever want to reuse these string and it adds an extra layer of unnecessary complexity.

Reply:
I think we should do that in a separate iteration. Though this URL is not used anywhere else, so we could move it here, but I think it would result in additional confusion whenever someone checks the file containing the routes, and not seeing all routes in there.